rcongui/src/components/PlayersHistory/index.js

```diff
@@ -281,7 +281,15 @@ const FilterPlayer = ({classes, constPlayersHistory, pageSize, total, page, setP
                             blacklisted={player.blacklisted}
                             flags={List(player.flags.map(v => Map(v)))}
                             onflag={() => setDoFlag(player)}
-                            onBlacklist={() => setDoConfirmPlayer({ player: player.steam_id_64, actionType: "blacklist" })}
+                            onBlacklist={
+                              () => setDoConfirmPlayer(
+                                {
+                                  player: player.steam_id_64,
+                                  actionType: "blacklist",
+                                  name: player.names ? player.names[player.names.length - 1] : null
```

Comment:
Why arbitrarily picking the last name here?

Reply:
By adding the name we can check if the blacklisted player is currently in the server. I assumed that the last name was the most recent one. We could ban the player if in game using just the steam id but it seems like that would be expensive as it would require repetitive calls to `get_player_info` (unless we already have a very recent cache for that?).

Reply:
No unfortunately the last name is not necessarily the most recent one. For example if:
1. Connects as "foo"
2. Connects as "bar"
3 Connects as "foo"

Bar will still be the last one, actually that's an issue for going stateless as well, as we don't have the link between player_sessions and player_names, that's something we'd need to add actually (in [save_session](https://gitlab.trading.imc.intra/amsbod/compliance/six/-/blob/master/six_reporting/modules/trade_selector/tas_utils.py)

So bottomline, yes it'd be better with steam ids. If you use the Rcon.get_players. ids are in there and are cached so it's not so expensive. (Please not that occasionnaly there's a bug where HLL does not return the steamid, rare but happens)

Reply:
I see. I think this is only needed if we want to directly ban a blacklisted player who is currently in the server, but it seems to be more trouble than it's worth so I will strip it out.

Reply:
I see. I think this is only needed if we want to directly ban a blacklisted player who is currently in the server, but it seems to be more trouble than it's worth so I will strip it out.